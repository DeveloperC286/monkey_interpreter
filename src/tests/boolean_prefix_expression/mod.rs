use super::*;

#[template]
#[rstest(
    code,
    snapshot_name,
    case("!false", "boolean_prefix_expression_case_1"),
    case("\t!TRUE", "boolean_prefix_expression_case_2"),
    case("  !!TRUE", "boolean_prefix_expression_case_3")
)]
fn boolean_prefix_expression_cases(code: &str, snapshot_name: &str) {}

#[apply(boolean_prefix_expression_cases)]
fn test_boolean_prefix_expression_lexical_analysis(code: &str, snapshot_name: &str) {
    assert_lexical_analysis!(code, snapshot_name);
}

#[apply(boolean_prefix_expression_cases)]
fn test_boolean_prefix_expression_syntax_analysis(code: &str, snapshot_name: &str) {
    assert_syntax_analysis!(code, snapshot_name);
}

#[apply(boolean_prefix_expression_cases)]
fn test_boolean_prefix_expression_evaluator(code: &str, snapshot_name: &str) {
    assert_evaluator!(code, snapshot_name);
}

#[apply(boolean_prefix_expression_cases)]
fn test_boolean_prefix_expression_environment(code: &str, snapshot_name: &str) {
    assert_environment!(code, snapshot_name);
}
