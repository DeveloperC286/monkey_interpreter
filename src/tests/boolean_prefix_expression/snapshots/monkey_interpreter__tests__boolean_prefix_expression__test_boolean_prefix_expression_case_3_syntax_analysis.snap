---
source: src/tests/boolean_prefix_expression/mod.rs
expression: abstract_syntax_tree
---
[
    Expression {
        expression: NotPrefix {
            right_hand: NotPrefix {
                right_hand: Boolean {
                    literal: true,
                },
            },
        },
    },
]
